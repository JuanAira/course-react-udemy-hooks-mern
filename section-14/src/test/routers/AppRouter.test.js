import { AppRouter } from '../../routers/AppRouter';
import { mount } from 'enzyme'
import { AuthContext } from '../../auth/authContext';


describe('Test in <AppRouter />', () => {
    test('should show login if not authenticate', () => {
        const contextValue = {
            user: {
                logged: false
            }
        };

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h1').text().trim()).toBe('Login');
    });

    test('should show component Marvel if is athenticate', () => {
        const contextValue = {
            user: {
                logged: true,
                name: 'Pepe'
            }
        };

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.navbar').exists()).toBe(true);
    });
});
