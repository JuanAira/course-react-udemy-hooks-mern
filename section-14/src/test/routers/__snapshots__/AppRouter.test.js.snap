// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AppRouter /> should show component Marvel if is athenticate 1`] = `ReactWrapper {}`;

exports[`Test in <AppRouter /> should show login if not authenticate 1`] = `ReactWrapper {}`;
